platform_plugin_teams/api/lms/views.py

```diff
@@ -0,0 +1,343 @@
+        field_errors = {}
+
+        team_id = request.data.get("team_id")
+        usernames = request.data.get("usernames")
+
+        if not team_id:
+            field_errors["team_id"] = "The [team_id] query parameter is required."
+
+        if not usernames:
+            field_errors["usernames"] = "The [usernames] query parameter is required."
+
+        if field_errors:
+            return api_field_errors(field_errors)
+
+        team = get_team_by_team_id(team_id=team_id)
+        if not team:
+            return api_field_errors(
+                {"team_id": f"The supplied {team_id=} does not exists."},
+                status_code=status.HTTP_404_NOT_FOUND,
+            )
+
+        if not has_specific_team_access(request.user, team):
+            return api_error(
+                f"The {request.user=} do not have access to the specified team.",
+                status_code=status.HTTP_403_FORBIDDEN,
+            )
+
+        if not can_user_modify_team(request.user, team):
+            return api_error(
+                f"The {request.user=} can't join an instructor managed team.",
+                status_code=status.HTTP_403_FORBIDDEN,
+            )
+
+        course_block = modulestore().get_course(team.course_id)
+
+        max_team_size = course_block.teams_configuration.calc_max_team_size(
+            team.topic_id
+        )
+        if (
+            max_team_size is not None
+            and team.users.exclude(username__in=usernames).count() + len(usernames)
+            > max_team_size
+        ):
+            return api_error(
+                f"The {team_id=} does not have enough space for the given users."
+            )
+
+        memberships = []
+
+        for username in usernames:
+            if not has_team_api_access(
+                request.user, team.course_id, access_username=username
+            ):
+                return api_error(
+                    (
+                        f"The {request.user=} do not have access "
+                        "to the Team API for the given course."
+                    ),
+                    status_code=status.HTTP_403_FORBIDDEN,
+                )
+
+            try:
+                user = get_user_by_username_or_email(username)
+            except User.DoesNotExist:
+                return api_field_errors(
+                    {"usernames": f"The {username=} does not exists."},
+                    status_code=status.HTTP_404_NOT_FOUND,
+                )
+
+            try:
+                membership = team.add_user(user)
+            except AlreadyOnTeamInTeamset:
+                old_membership = CourseTeamMembership.objects.filter(
+                    user=user,
+                    team__course_id=team.course_id,
+                    team__topic_id=team.topic_id,
+                ).first()
+                old_membership.delete()
+                membership = CourseTeamMembership.objects.create(user=user, team=team)
```

Comment:
Same question here

Reply:
Here we can use the same method to create the membership: `team.add_user(user)`